src/assets/ReasonAsset.js

```diff
@@ -0,0 +1,28 @@
+const bsb = require('bsb-js');
+const fs = require('fs');
+const JSAsset = require('./JSAsset');
+const promisify = require('../utils/promisify');
+const readFile = promisify(fs.readFile);
+
+class ReasonAsset extends JSAsset {
+  async parse(code) {
+    // This runs BuckleScript - the Reason to JS compiler.
+    // Other Asset types use `localRequire` but the `bsb-js` package already
+    // does that internally. This should also take care of error handling in
+    // the Reason compilation process.
+    if (process.env.NODE_ENV !== 'test') {
+      await bsb.runBuild();
+    }
+
+    // This is a simplified use-case for Reason - it only loads the recommended
+    // BuckleScript configuration to simplify the file processing.
+    const outputFile = this.name.replace(/\.(re|ml)$/, '.bs.js');
+    const outputContent = await readFile(outputFile);
```

Comment:
is there no way to do this without writing to a temporary file first?

Reply:
Sorry, what do you mean by writing to a temp file?

Reply:
I mean, it seems like `bsb-js` or whatever that uses is writing its output to a temp file, which we then need to read here. Is there a way to avoid that?

Reply:
e.g. the `compileFile` function as used by parcel-plugin-bucklescript seems to return the output directly to a variable

Reply:
compileFile basically does the same thing here — reading the result of the output. Currently BuckleScript has no way of compiling Reason like that without seriously compromised performance and maintainability.